cinder/scheduler/host_manager.py

```diff
@@ -517,6 +522,119 @@ def get_weighed_backends(self, backends, weight_properties,
+                    for tmp_agg_id in _agg_states[tmp_back]:
+                        if tmp_agg_id == agg_id:
+                            _caps = _agg_states[tmp_back][tmp_agg_id]
+                            orig_caps['allocated_capacity_gb'] += \
+                                _caps['allocated_capacity_gb']
+
+        # Save the relevant stats from the original capabilities
+        # {<backend>: {<agg_id>: {allocated_capacity_gb: 0,
+        #                         pool_name: 'pool_name'}}}
+        self.aggregate_service_states[backend] = _agg_caps
+        return capabilities
+
+    def consume_from_volume_aggregate(self, backend, size):
+        """Update all the pools that match an aggregate.
+
+        This will go through all the backends that have the same aggregate
+        and update the state of the pool allocated_capacity_gb.
+
+        This is to ensure all the pool stats for the same aggregate
+        id are up to date between each call to update_service_capabilities,
+        which happens about every 60 seconds or so.
+
+        The backend passed in here has a pool assigned to the host, so we
+        know which pool that just got new provisioned space against it.
+        """
+        current_backend = volume_utils.extract_host(backend.host, 'backend')
+        if current_backend not in self.aggregate_service_states:
+            # We don't have any aggregate pools to update
+            return
+
+        current_backend_host = backend.host
+        current_pool = volume_utils.extract_host(backend.host, 'pool')
+        current_agg_id = None
+
+        # Now look for an aggregate_id in the capabilities of the pool
+        if current_backend in self.backend_state_map:
+            backend_state = self.backend_state_map[current_backend]
+            caps = backend_state.pools[current_pool].capabilities
+            if "aggregate_id" in caps:
+                current_agg_id = caps['aggregate_id']
+
+        if not current_agg_id:
+            return
+
+        # we know we have an aggregate_id, so now we need to loop through
+        # all other backends and update the allocated_capacity for all the
+        # same pools with the same aggregate id.
+        for backend_key, state in self.backend_state_map.items():
+            for key in state.pools:
+                pool_state = state.pools[key]
+                if current_backend_host == "#".join([backend_key,
+                                                     pool_state.pool_name]):
+                    continue
+
+                agg_id = pool_state.capabilities.get('aggregate_id')
+                if agg_id == current_agg_id:
+                    LOG.info("Consuming %(size)s from '%(backend_key)s'"
+                             "pool '%(pool_name)s' aggregate_id "
+                             "'%s(current_agg_id)s'",
+                             {'size': size, 'pool_name': pool_state.pool_name,
+                              'aggregate_id': current_agg_id})
```

Comment:
`backend_key` is missing as paraemter.

there's a typo in `'%s(current_agg_id)s'` - the `s` in front needs to go. Additionally, the key to the log call is called `aggregate_id` and not `current_agg_id`.

Reply:
Done